src/clipping/weileratherton.jl

```diff
@@ -0,0 +1,408 @@
+    if !isnothing(k)
+      ring = ring[(k + 1):end]
+    end
+
+    if length(ring) > 2
+      push!(rings, Ring([r.point for r in ring]...))
+    end
+  end
+  return rings
+end
+
+function decide_direction(clipped₁, clipped₂, clipping₁, clipping₂)
+  # The input segments should all share one common vertex.
+  # Form vectors from the common vertex to other vertices.
+  a = minimum(clipped₁) - maximum(clipped₁)
+  b = maximum(clipped₂) - minimum(clipped₂)
+  c = minimum(clipping₁) - maximum(clipping₁)
+  d = maximum(clipping₂) - minimum(clipping₂)
+
+  α = mod(atan(a[2], a[1]), 2pi) # Angle formed by the clipped₁ segment and a horizontal line.
+  β = mod(atan(b[2], b[1]) - α, 2pi) # Angle between clipped₁ and clipped₂.
+  γ = mod(atan(c[2], c[1]) - α, 2pi) # Angle between clipped₁ and clipping₁.
+  δ = mod(atan(d[2], d[1]) - α, 2pi) # Angle between clipped₁ and clipping₂.
+
+  if isapprox(γ, 0.0) || isapprox(γ, 2pi)
+    if δ < β < 2pi
+      return Entering
+    else
+      return Exiting
+    end
+  elseif isapprox(δ, 0.0) || isapprox(δ, 2pi)
+    if γ < β < 2pi
+      return Exiting
+    else
+      return Entering
+    end
+  end
+  if γ < δ && (γ < β || isapprox(β, γ)) && (isapprox(β, δ) || β < δ)
+    return Exiting
+  elseif δ < γ && (δ < β || isapprox(β, δ)) && (isapprox(β, γ) || β < γ)
+    return Entering
+  end
+  return nothing
+end
+
+function insertintersections!(clipping, clipped, point, vtype, entering)
+  intersection = RingVertex{vtype}(point)
+  insertintersection!(clipping, intersection, :left)
+  insertintersection!(clipped, intersection, :right)
+
+  if vtype == Entering
+    push!(entering, intersection)
+  end
+end
+
+function clip(poly::Polygon, ring::Ring, _::WeilerAthertonClipping)
+  polyrings = rings(poly)
+
+  # Convert the subject polygon rings and the clipping ring to the RingVertex data structure.
+  clippedrings = [gettraversalring(r) for r in polyrings]
+  startclipping = gettraversalring(ring)
+
+  # For keeping track of intersected rings, as the non-intersected ones need additional
+  # processing at the end.
+  intersected = zeros(Bool, length(clippedrings))
+
+  # For collecting all entering vertices, as they are used as starting points for collection
+  # of the output rings.
+  entering::Vector{RingVertex{Entering}} = []
+
+  clipping = startclipping
+  while true
+    # Three consecutive clipping vertices are used to properly identify intersection vertex type
+    # for corner cases, so the clipping segment is constructed with the two following vertices
+    # after the current one.
+    clippingsegment = Segment(clipping.left.point, clipping.left.left.point)
+
+    for (k, startclipped) in enumerate(clippedrings)
+      clipped = startclipped
+      while true
+        # Like for the clipping, the clipped also uses three consecutive vertices.
+        clippedsegment = Segment(clipped.left.point, clipped.left.left.point)
+
+        vertex_type = nothing
```

Comment:
still not full adherence to code style, we don't use underscore unless strictly needed

Reply:
Now this should be corrected everywhere.